dvc/repo/__init__.py

```diff
@@ -280,7 +280,7 @@ def _ignore(self):
 
         self.scm.ignore_list(flist)
 
-    def check_modified_graph(self, new_stages):
+    def check_modified_graph(self, new_stages, old_stages=None):
```

Comment:
Does it make sense to have "old_stages"? Since `repo.stages` are called in `validate_state`, won't they be already cached here?

Reply:
Yes, they will be cached if `force=True` in the `validate_state`. And, in that case, we are (force) removing an existing stage with the new one in a new (shallow) copy of `repo.stages` to check for the graph. 

Otherwise, `repo.stages` is used if `force=False`.